SingularityService/src/main/java/com/hubspot/singularity/hooks/LoadBalancerClientImpl.java

```diff
@@ -68,6 +72,40 @@ public LoadBalancerClientImpl(SingularityConfiguration configuration, ObjectMapp
     this.mesosProtosUtils = mesosProtosUtils;
   }
 
+  private String getStateUriFromRequestUri (){
+    return loadBalancerUri.replace("request", "state");
+  }
+
+  private String getLoadBalancerStateUri (String requestId){
+    return String.format(OPERATION_URI, getStateUriFromRequestUri(), requestId);
+  }
+
+  private Optional<BaragonServiceState> getBaragonServiceStateForLoadBalancerRequest (String requestId) {
+    final String loadBalancerStateUri = getLoadBalancerStateUri(requestId);
+    final BoundRequestBuilder requestBuilder = httpClient.prepareGet(loadBalancerStateUri);
+    final Request request = requestBuilder.build();
+    try {
+      ListenableFuture<Response> future = httpClient.executeRequest(request);
+      Response response = future.get(loadBalancerTimeoutMillis, TimeUnit.MILLISECONDS);
+      if (!JavaUtils.isHttpSuccess(response.getStatusCode())) {
+        LOG.info(String.format("Response status code %s", response.getStatusCode()));
+      }
+      return objectMapper.readValue(response.getResponseBodyAsBytes(), new TypeReference<BaragonServiceState>() {});
+    } catch (IOException | InterruptedException | ExecutionException | TimeoutException e) {
+      e.printStackTrace();
+      return Optional.absent();
+    }
+  }
+
+  public Collection<UpstreamInfo> getBaragonUpstreamsForRequest (String requestId) {
+    Optional<BaragonServiceState> maybeBaragonServiceState = getBaragonServiceStateForLoadBalancerRequest(requestId);
+    if (maybeBaragonServiceState.isPresent()){
+      BaragonServiceState baragonServiceState = getBaragonServiceStateForLoadBalancerRequest(requestId).get();
```

Comment:
don't need to call `getBaragonServiceStateForLoadBalancerRequest` twice here, which will make the http call twice. This can just be `maybeBaragonServiceState.get()`